### PR TITLE
LGFS hardship tweaks

### PR DESCRIPTION
**Pull request by colinbruce**
#### What
Fix feature displaying case_stage instead of case_type
Update case_stage hint text, sallow agfs and lgfs examples
Update the case stage view with error handling

### CODE REVIEW

File: app/views/external_users/claims/case_details/_case_type_fields.html.haml [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
+    %label.form-label-bold{ for: 'case_stage' }
       = t('.stage_type')
       .form-hint.xsmall.zero-vert-margin
-        = t('.stage_type_hint')
-      = validation_error_message(@error_presenter, :case_type)
+        = t(".#{@claim.agfs? ? 'agfs' : 'lgfs'}_stage_type_hint")
+      = validation_error_message(@error_presenter, :case_stage)
 
-    = f.select :case_stage_id, @case_stages.map{ |cs| [cs.description, cs.id, { data: { 'is-fixed-fee': cs.is_fixed_fee?, 'requires-cracked-dates': cs&.requires_cracked_dates?, 'requires-retrial-dates': cs&.requires_retrial_dates?, 'requires-trial-dates': cs&.requires_trial_dates? } }] }, { include_blank: ''.html_safe }, { class: 'form-control fx-autocomplete', id: 'case_type', 'aria-label': t('.stage_type') }
+    = f.select :case_stage_id, @case_stages.map{ |cs| [cs.description, cs.id, { data: { 'is-fixed-fee': cs.is_fixed_fee?, 'requires-cracked-dates': cs&.requires_cracked_dates?, 'requires-retrial-dates': cs&.requires_retrial_dates?, 'requires-trial-dates': cs&.requires_trial_dates? } }] }, { include_blank: ''.html_safe }, { class: 'form-control fx-autocomplete', id: 'case_stage', 'aria-label': t('.stage_type') }
```
Comment (jsugarman) [inline app/views/external_users/claims/case_details/_case_type_fields.html.haml]:
```suggestion
    = f.select :case_stage_id, @case_stages.map{ |cs| [cs.description, cs.id, { data: { 'is-fixed-fee': cs.is_fixed_fee?, 'requires-cracked-dates': cs&.requires_cracked_dates?, 'requires-retrial-dates': cs&.requires_retrial_dates?, 'requires-trial-dates': cs&.requires_trial_dates? } }] }, { include_blank: ''.html_safe }, { class: 'form-control fx-autocomplete', id: 'case_type', 'aria-label': t('.stage_type') }
```

using case_stage as id, breaks the js hook for displaying trial/retrial/cracked trial fields

Comment (colinbruce) [reply app/views/external_users/claims/case_details/_case_type_fields.html.haml]: That's what links the error without an ID named case_stage, the error message will not link

Comment (jsugarman) [reply app/views/external_users/claims/case_details/_case_type_fields.html.haml]: i imagine/hope tests will fail I think the other changes should be sufficient. however not using the case_stage as id breaks error linking

you could use a hacky fix for the error linking but there are issues with this too
```
   #case_stage
      = f.select :case_stage_id, @case_stages.map{ |cs| [cs.description, cs.id, { data: { 'is-fixed-fee': cs.is_fixed_fee?, 'requires-cracked-dates': cs&.requires_cracked_dates?, 'requires-retrial-dates': cs&.requires_retrial_dates?, 'requires-trial-dates': cs&.requires_trial_dates? } }] }, { include_blank: ''.html_safe }, { class: 'form-control fx-autocomplete', id: 'case_type', 'aria-label': t('.stage_type') }
```

Comment (colinbruce) [reply app/views/external_users/claims/case_details/_case_type_fields.html.haml]: Trying something else... see the next commit

Comment (jsugarman) [reply app/views/external_users/claims/case_details/_case_type_fields.html.haml]: cool, that will do for now 👍 @naseberry perhaps you can look at this in more detail.

**Review by jsugarman — APPROVED** [review]
👍

File: app/views/external_users/claims/case_details/_case_type_fields.html.haml [hunk 2]
```diff
@@ -1,11 +1,10 @@
 - if @claim.hardship?
-  #cc-case-type.form-group{ class: error_class?(@error_presenter, :case_type) ? 'form-group-error dropdown_field_with_errors' : '' }
-    %label.form-label-bold{ for: 'case_type' }
+  #cc-case-stage.form-group{ class: error_class?(@error_presenter, :case_stage) ? 'form-group-error dropdown_field_with_errors' : '' }
```
Comment (jsugarman) [inline app/views/external_users/claims/case_details/_case_type_fields.html.haml:2]: cc-case-type
```suggestion
  #cc-case-type.form-group{ class: error_class?(@error_presenter, :case_stage) ? 'form-group-error dropdown_field_with_errors' : '' }
```

**Pull request merged**